plugins/woo-ai/api/text-generation/class-text-generation.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * REST API Data Controller
+ *
+ * Handles requests to /data
+ *
+ * @package Woo_AI
+ */
+
+namespace Automattic\WooCommerce\Admin\API;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Data controller.
+ *
+ * @internal
+ * @extends WC_REST_Data_Controller
+ */
+class Text_Generation extends \WC_REST_Data_Controller {
+
+	/**
+	 * Model details.
+	 *
+	 * @var array
+	 */
+	protected $models = array(
+		'openai' => array(
+			'url' => 'https://api.openai.com/v1/chat/completions',
+			'key' => OPEN_AI_KEY,
+		),
+	);
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->register_routes();
+	}
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wooai';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'text-generation';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'get_response' ),
+					'permission_callback' => array( $this, 'get_response_permission_check' ),
+					'args'                => array(
+						'prompt'            => array(
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+							'required'          => true,
+						),
+						'temperature'       => array(
+							'type'              => 'number',
+							'validate_callback' => 'rest_validate_request_arg',
+						),
+						'previous_messages' => array(
+							'description' => __( 'Optional parameter to get only specific task lists by id.', 'woocommerce' ),
+							'type'        => 'array',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check if a given request has access to read.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_response_permission_check( $request ) {
+		if ( ! wc_rest_check_post_permissions( 'product', 'create' ) ) {
+			return new \WP_Error( 'woocommerce_rest_cannot_create', __( 'Sorry, you are not allowed to create resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the onboarding tasks.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_response( $request ) {
+		$prompt            = $request->get_param( 'prompt' );
+		$temperature       = $request->get_param( 'temperature' ) ? $request->get_param( 'temperature' ) : 1;
+		$previous_messages = $request->get_param( 'previous_messages' ) ? $request->get_param( 'previous_messages' ) : array();
+		$request_model     = $request->get_param( 'model' ) ? $request->get_param( 'model' ) : 'openai';
```

Comment:
model doesn't actually look like an allowed param for this endpoint, should we leave this out for now and just set it to `openai` to reduce complexity?

Reply:
Sounds good, simplified in https://github.com/woocommerce/woocommerce/pull/38309/commits/f53bde54458a34492beea69974d67f1704c7fc0c.